test/cpp_extensions/cudnn_extension.cpp

```diff
@@ -0,0 +1,73 @@
+#include <torch/torch.h>
+#include <cuda.h>
+#include <torch/torch.h>
+#include <ATen/ATen.h>
+#include <ATen/TensorUtils.h>
+#include <ATen/cudnn/Descriptors.h>
+#include <ATen/cudnn/Exceptions.h>
+#include <ATen/cudnn/Handles.h>
+#include <torch/csrc/cuda/cuda_check.h>
+#include <torch/torch.h>
+#include <iostream>
+
+using namespace at;
+using namespace std;
+using namespace at::native;
+const char *cudnn_relu_name = "cudnn_relu";
+
+// Check arguments to cudnn_relu
+int cudnn_relu_check(const Tensor &inputs, const Tensor &outputs) {
+  // Create TensorArgs
+  TensorArg arg_inputs(inputs, "inputs", 0);
+  TensorArg arg_outputs(outputs, "outputs", 1);
+  // Check arguments
+  checkContiguous(cudnn_relu_name, arg_inputs);
+  checkScalarType(cudnn_relu_name, arg_inputs, ScalarType::Float);                      \
+  checkBackend(cudnn_relu_name, arg_inputs.tensor, Backend::CUDA);
+  checkContiguous(cudnn_relu_name, arg_outputs);
+  checkScalarType(cudnn_relu_name, arg_outputs, ScalarType::Float);                      \
+  checkBackend(cudnn_relu_name, arg_outputs.tensor, Backend::CUDA);
+  checkSameSize(cudnn_relu_name, arg_inputs, arg_outputs);
+  return 0;
+}
+
+int cudnn_relu(const Tensor& inputs, const Tensor& outputs) {
+  //Check inputs
+  cudnn_relu_check(inputs, outputs);
+  // Declarations
+  cudnnHandle_t cuDnn = getCudnnHandle();
+  TensorDescriptor* input_tensor_desc = new TensorDescriptor(inputs, 4);
+  // 4 is minium size for TensorDescriptor
+  cudnnActivationDescriptor_t activationDesc;
+  cudnnActivationMode_t mode = CUDNN_ACTIVATION_RELU;
+  cudnnNanPropagation_t reluNanOpt = CUDNN_PROPAGATE_NAN;
+  double coef = 1.;
+  float alpha = 1.;
+  float beta = 0.;
+  // Activation descriptor
+  CUDNN_CHECK(cudnnCreateActivationDescriptor(&activationDesc));
+  CUDNN_CHECK(cudnnSetActivationDescriptor(
+      activationDesc,
+      mode,
+      reluNanOpt,
+      coef));
+  //Apply
```

Comment:
Not sure this kind of comment is actually useful at all

Reply:
More for visually breaking things up than information, but I can toss it

Reply:
Actually, I was trying to get across that most cudnn extensions are going to follow the same basic structure:
- create descriptors
- set descriptors
- apply
- destroy descriptors
- return

I'll try to make that more clear in the comments

Reply:
Yeah, I think if you could explain more (at a higher level) what is going on this will be very helpful for folks looking to use cuDNN (maybe with little/no experience) in C++ extensions.